src/otg/finngen.py

```diff
@@ -0,0 +1,69 @@
+"""Step to run FinnGen study table ingestion."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from urllib.request import urlopen
+
+from pyspark.sql import functions as f
+
+from otg.common.session import Session
+from otg.config import FinnGenStepConfig
+
+
+@dataclass
+class FinnGenStep(FinnGenStepConfig):
+    """FinnGen study table ingestion step.
+
+    The following information is aggregated/extracted:
+    - Study ID in the special format (FINNGEN_R7_*)
+    - Trait name (for example, Amoebiasis)
+    - Number of cases and controls
+    - Link to the summary statistics location
+
+    Some fields are also populated as constants, such as study type and the initial sample size.
+    """
+
+    session: Session = Session()
+
+    def run(self: FinnGenStep) -> None:
```

Comment:
I think it is not necessarily a blocker for merging the PR, but the organization of the logic in PR doesn't follow the existing implementation. Currently we have datasets, eg. [StudyIndex](https://github.com/opentargets/genetics_etl_python/blob/bff5c0d0c2124e7f89ad132d13f375fb891c5bea/src/otg/dataset/study_index.py#L28) whit methods to generate the dataset (and brings the logic to validate data) or there are other [classes](https://github.com/opentargets/genetics_etl_python/blob/bff5c0d0c2124e7f89ad132d13f375fb891c5bea/src/otg/dataset/study_index.py#L60) that inherit from the dataset. While the "run" scripts are just wrapper around these methods. Up to you @d0choa, @ireneisdoomed to request for change or not.

Reply:
Thank you for pointing this out Daniel! This has now all been addressed

Reply:
The circular dependency doesn't exist for other sources but the GWAS Catalog. It's due to the curation practices.